### PR TITLE
Implement random respawning

### PR DESCRIPTION
**Pull request by nabijaczleweli**
Checks for collisions with bullets and players.
Also moves server stuff to `server` module, cleanliness.

Beef is https://github.com/LoungeCPP/Tatsoryk/compare/b8d3df320b95fc43e1d06a51269cc9db28ea1339...681fa68cf1999a1242c4152e8d96fe3ffd102883, rest is "noise".

### CODE REVIEW

File: server/src/server/gamestate.rs [hunk 1]
```diff
@@ -183,4 +185,32 @@ impl GameState {
             _ => panic!("Unprocessed message! {}", message.to_string()),
         }
     }
+
+    fn random_free_spot<R: Rng>(&self, rng: &mut R) -> (f32, f32) {
+        loop {
```
Comment (TheCatPlusPlus) [inline server/src/server/gamestate.rs]: Needs some limit on iteration count, so we don't loop endlessly in the worst case.

Comment (nabijaczleweli) [inline server/src/server/gamestate.rs]: Done

File: server/src/server/gamestate.rs [hunk 2]
```diff
@@ -183,4 +186,38 @@ impl GameState {
...
+                }
+            }
+
+            if !collides {
+                return (x, y);
+            }
+        }
+        println!("Failed to find a random empty spot for player after {} iterations", MAX_ITERATIONS);
+
+        // Hardcoded map boundaries
```
Comment (TheCatPlusPlus) [inline server/src/server/gamestate.rs]: Constant maybe then?

Comment (nabijaczleweli) [inline server/src/server/gamestate.rs]: Done

**Comment by TheCatPlusPlus** [comment]
:+1:

**Pull request merged**